libraries/AP_Vehicle/AP_Vehicle.cpp

```diff
@@ -205,6 +205,10 @@ void AP_Vehicle::setup()
     // initialise serial ports
     serial_manager.init();
     gcs().setup_console();
+#ifndef HAL_BUILD_AP_PERIPH
```

Comment:
```suggestion
#if AP_SERIALMANAGER_CONFIGURATION_CHECKING_ENABLED
```

default that to on in `AP_SerialManager_config.h`.  Turn it off in the `ap_periph-defaults` method in `chibios_hwdef.py` - and if you're keen back on again for the "heavy" periphs that actually support GCSs.

Gate all the code in AP_SerialManager with the same define.

Reply:
Fixed